packages/kbn-content-management/table_list/src/table_list_view/components/listing_limit_warning.tsx

```diff
@@ -0,0 +1,80 @@
+/*
+ * Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
+ * or more contributor license agreements. Licensed under the Elastic License
+ * 2.0 and the Server Side Public License, v 1; you may not use this file except
+ * in compliance with, at your election, the Elastic License 2.0 or the Server
+ * Side Public License, v 1.
+ */
+/* eslint react-hooks/exhaustive-deps: 2 */
+
+import React from 'react';
+import { EuiCallOut, EuiLink, EuiSpacer } from '@elastic/eui';
+import { FormattedMessage } from '@kbn/i18n-react';
+
+interface Props<T> {
+  entityNamePlural: string;
+  canEditAdvancedSettings: boolean;
+  advancedSettingsLink: string;
+  totalItems: number;
+  listingLimit: number;
+}
+
+export function ListingLimitWarning<T>({
+  entityNamePlural,
+  totalItems,
+  listingLimit,
+  canEditAdvancedSettings,
+  advancedSettingsLink,
+}: Props<T>) {
+  return (
+    <>
+      <EuiCallOut
+        title={
+          <FormattedMessage
+            id="contentManagement.tableList.listing.listingLimitExceededTitle"
+            defaultMessage="Listing limit exceeded"
+          />
+        }
+        color="warning"
+        iconType="help"
+      >
+        <p>
+          {canEditAdvancedSettings ? (
+            <FormattedMessage
```

Comment:
This is one of the reasons I avoid `<FormattedMessage>`-- this is really hard to grok.

Reply:
I used it so much that my eyes got used to it probably 😄 In this file we _do_ need it though as we render `<EuiLink>` and `<strong>` DOM element inside the translation.